cpp/src/arrow/dataset/file_orc.cc

```diff
@@ -85,24 +85,20 @@ class OrcScanTask : public ScanTask {
           included_fields.push_back(name);
         }
 
+        std::shared_ptr<RecordBatchReader> recordBatchReader;
+        reader->NextStripeReader(scan_options.batch_size, included_fields, &recordBatchReader);
```

Comment:
I am not sure if using the `NextStripeReader` is correct here, see the discussion about this on the original PR adding ORC dataset support: https://github.com/apache/arrow/pull/10991#discussion_r698447362 (I initially also used `NextStripeReader`, but so the name is a bit confusing, as it doesn't iteratively reads the "next" stripe, it only reads batches from a single stripe)

Reply:
@jorisvandenbossche 

Recently, our testing found that the NextStripeReader  only read first stripe of the big files. I will fix this as soon as possible.

Is anyone fixing this issue? We Can discuss and solve this problem together.

Thanks

Reply:
@jorisvandenbossche @pitrou 

Have fixed the stripes iteration for batch_size reading, please have a review.

Thanks!